libcaf_core/test/detail/parser/read_number.cpp

```diff
@@ -269,4 +301,44 @@ CAF_TEST(fractional mantissa with negative exponent) {
   CHECK_NUMBER(-42.0001e-5);
 }
 
+#define CHECK_RANGE(expr, ...)                                                 \
+  CAF_CHECK_EQUAL(r(expr), std::vector<int64_t>({__VA_ARGS__}))
+
+CAF_TEST(a range from n to n is just n) {
+  CHECK_RANGE("0..0", 0);
+  CHECK_RANGE("1..1", 1);
+  CHECK_RANGE("2..2", 2);
+  CHECK_RANGE("101..101", 101);
+  CHECK_RANGE("101..101..1", 101);
+  CHECK_RANGE("101..101..2", 101);
+  CHECK_RANGE("101..101..-1", 101);
+  CHECK_RANGE("101..101..-2", 101);
+}
+
+CAF_TEST(ranges are either ascending or descending) {
+  CHECK_RANGE("0..1", 0, 1);
+  CHECK_RANGE("0..2", 0, 1, 2);
+  CHECK_RANGE("0..3", 0, 1, 2, 3);
+  CHECK_RANGE("3..0", 3, 2, 1, 0);
+  CHECK_RANGE("3..1", 3, 2, 1);
+  CHECK_RANGE("3..2", 3, 2);
+}
+
+CAF_TEST(ranges can use positive step values) {
+  CHECK_RANGE("2..6..2", 2, 4, 6);
+  CHECK_RANGE("3..8..3", 3, 6);
+}
+
+CAF_TEST(ranges can use negative step values) {
+  CHECK_RANGE("6..2..-2", 6, 4, 2);
+  CHECK_RANGE("8..3..-3", 8, 5);
+}
+
+CAF_TEST(ranges can use signed integers) {
+  CHECK_RANGE("+2..+6..+2", 2, 4, 6);
+  CHECK_RANGE("+6..+2..-2", 6, 4, 2);
+  CHECK_RANGE("+2..-2..-2", 2, 0, -2);
+  CHECK_RANGE("-2..+2..+2", -2, 0, 2);
```

Comment:
What happens for a range like this: `"-2..+2..-2"`?

Reply:
You'll get an error. I've added a new test case for that.